### PR TITLE
landing page

### PR DESCRIPTION
**Pull request by vaibhavchalse99**
Added API integration of landing page to validate the token, to set authToken using redux and saga

### CODE REVIEW

File: src/actions/userDriveActions.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { DRIVE } from "constants/actionConstants";
+
+export const driveDetailRequest = (value) => {
```
Comment (shailendrakanherkar18) [inline src/actions/userDriveActions.js:3]: export const driveDetailRequest = (value) => ({
    type: DRIVE.DRIVE_DETAIL_REQUEST,
    payload: { token: value },
  });
  
  Single statement can be returned directly

File: src/actions/userDriveActions.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  return {
+    type: DRIVE.SET_DETAILS,
+    payload: value,
+  };
+};
+
+export const showErrorMessage = (value) => {
+  return {
+    type: DRIVE.SET_ERROR_MESSAGE,
+    payload: { error: true, errorMessage: value },
```
Comment (shailendrakanherkar18) [inline src/actions/userDriveActions.js:20]: errorMessage: value
parameter should be rename with the proper name

File: src/actions/userProfileActions.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import { USER } from "constants/actionConstants";
+
+export const setUserProfileDetails = (value) => {
+  return {
+    type: USER.SET_DETAILS,
+    payload: value,
```
Comment (shailendrakanherkar18) [inline src/actions/userProfileActions.js:6]: payload: value, to be on consistency side , will add values inside payload not to the payload.

File: src/apis/userDriveApis.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import axios from "axios";
+
+import { SERVER_URL } from "constants/appConstants";
+
+const driveDetail = (token) => {
+  return axios.get(`${SERVER_URL}/validate-token/${token}`);
```
Comment (shailendrakanherkar18) [inline src/apis/userDriveApis.js:6]: We can use axios.create to configure baseUrl

File: src/components/LandingPageComponent/index.jsx [hunk 5]
```diff
@@ -0,0 +1,51 @@
+import moment from "moment";
+
+import { Button, Alert } from "core-components";
+import Loading from "shared-components/Loading";
+
+function LandingPageComponent(props) {
+  const { startTime, error, errorMessage, loading, handleClick } = props;
```
Comment (shailendrakanherkar18) [inline src/components/LandingPageComponent/index.jsx:7]: Boolean flag should be prefix with is i.e isLoading

File: src/components/LandingPageComponent/index.jsx [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+      </div>
+    );
+  }
+
+  return (
+    <div className="overview-block text-center text-white">
+      <h3 className="mt-5">Welcome to</h3>
+      <h1
+        className="font-weight-bolder"
+        style={{
```
Comment (shailendrakanherkar18) [inline src/components/LandingPageComponent/index.jsx:26]: Do inline style unless it is a dynamic one

File: src/constants/actionConstants.js [hunk 7]
```diff
@@ -1 +1,10 @@
-// export const ACTION_CONSTANT = "ACTION_CONSTANTS";
```
Comment (shailendrakanherkar18) [inline src/constants/actionConstants.js:1]: remove unwanted code

File: src/constants/actionConstants.js [hunk 8]
```diff
@@ -1 +1,10 @@
...
+export const DRIVE = {
+  DRIVE_DETAIL_REQUEST: "DRIVE_DETAIL_REQUEST",
+  SET_DETAILS: "SET_DRIVE_DETAILS",
+  SET_ERROR_MESSAGE: "SET_ERROR_MESSAGE",
```
Comment (shailendrakanherkar18) [inline src/constants/actionConstants.js:4]: Each action should have its feature name
SET_ERROR_MESSAGE => SET_**DRIVE**_ERROR_MESSAGE

File: src/dbRules.json [hunk 9]
```diff
@@ -0,0 +1,32 @@
+{
```
Comment (shailendrakanherkar18) [inline src/dbRules.json:1]: If this is mock data, we can create a directory mock-json
and move files within it

File: src/reducers/userDriveReducer.js [hunk 10]
```diff
@@ -0,0 +1,29 @@
+import { DRIVE } from "constants/actionConstants";
```
Comment (shailendrakanherkar18) [inline src/reducers/userDriveReducer.js:1]: Can we add immer to achive immutibility ????

File: src/reducers/userDriveReducer.js [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+  error: false,
+  errorMessage: "",
+  loading: false,
+};
+
+const userDriveReducer = (state = initialState, action) => {
+  switch (action.type) {
+    case DRIVE.SET_DETAILS:
+      return { ...state, ...action.payload };
+
```
Comment (shailendrakanherkar18) [inline src/reducers/userDriveReducer.js:17]: remove blacnk lines

File: src/reducers/userDriveReducer.js (hunk 10 above)
Comment (shailendrakanherkar18) [inline src/reducers/userDriveReducer.js:1]: Can we add immer library to achive immutability

File: src/root/Routes.jsx [hunk 12]
```diff
@@ -13,8 +13,8 @@ function Routes() {
       <Route exact path="/">
```
Comment (shailendrakanherkar18) [inline src/root/Routes.jsx:13]: Can wedefine paths in separate constant file

File: src/sagas/userDriveSagas.js [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+import {
+  setUserDriveDetails,
+  showErrorMessage,
+  setDriveLoading,
+} from "actions/userDriveActions";
+
+//worker saga
+export function* driveDetails(action) {
+  try {
+    yield put(setDriveLoading(true));
```
Comment (shailendrakanherkar18) [inline src/sagas/userDriveSagas.js:15]: Can we manage this laoding flag within reducer???

File: src/sagas/userDriveSagas.js [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+    const { data } = yield call(driveDetail, action.payload.token);
+    const { userDetails, driveDetails, authToken } = data;
+    yield put(setDriveLoading(false));
+    yield put(setUserDriveDetails(driveDetails));
+    yield put(setUserProfileDetails(userDetails));
+    localStorage.setItem("authToken", authToken);
+  } catch (err) {
+    yield put(setDriveLoading(false));
```
Comment (shailendrakanherkar18) [inline src/sagas/userDriveSagas.js:23]: setDriveLoading this also withing reducer

File: src/components/LandingPageComponent/index.jsx (hunk 5 above)
Comment (vaibhavchalse99) [reply src/components/LandingPageComponent/index.jsx:7]: is this applicable for the redux store too?

**Review by Suhas010 — APPROVED** [review]

File: src/apis/apiHelper.js [hunk 15]
```diff
@@ -0,0 +1,21 @@
...
+
+import { SERVER_URL } from 'constants/appConstants';
+
+const client = axios.create({
+  baseURL: SERVER_URL,
+});
+
+const get = (path) => client.get(path);
+
+const post = (path, data) => client.post(path, data);
```
Comment (shailendrakanherkar18) [inline src/apis/apiHelper.js:11]: We can also keep provision for query params in post request .

File: src/containers/LandingPageContainer.jsx [hunk 16]
```diff
@@ -0,0 +1,37 @@
...
+  const history = useHistory();
+
+  const userDriveState = useSelector((state) => state.userDriveReducer);
+  const { startTime, isError, errorMessage, isLoading } = userDriveState;
+
+  useEffect(() => {
+    dispatch(driveDetailRequest(tokenId));
+  }, [dispatch, tokenId]);
+
+  const handleClick = () => {
```
Comment (shailendrakanherkar18) [inline src/containers/LandingPageContainer.jsx:22]: Each memeber function should be wrap with useCallback
And each functional component should be wrap with React.memo

File: src/reducers/userDriveReducer.js [hunk 17]
```diff
@@ -0,0 +1,34 @@
...
+  },
+  isError: false,
+  errorMessage: '',
+  isLoading: false,
+};
+
+const userDriveReducer = produce((state = initialState, action) => {
+  switch (action.type) {
+    case DRIVE.SET_DETAILS:
+      return { ...state, ...action.payload, isLoading: false };
```
Comment (shailendrakanherkar18) [inline src/reducers/userDriveReducer.js:20]: state.data = action.payload

File: src/sagas/userDriveSagas.js [hunk 18]
```diff
@@ -0,0 +1,28 @@
...
+} from 'actions/userDriveActions';
+
+// worker saga
+export function* driveDetails(action) {
+  try {
+    const { data } = yield call(driveDetail, action.payload.token);
+    const { userDetails, driveDetails: userDriveDetails, authToken } = data;
+    yield put(setUserDriveDetails(userDriveDetails));
+    yield put(setUserProfileDetails(userDetails));
+    localStorage.setItem('authToken', authToken);
```
Comment (shailendrakanherkar18) [inline src/sagas/userDriveSagas.js:19]: for localStorage utilities can we create wrapper

**Review by shailendrakanherkar18 — APPROVED** [review]

**Pull request merged**